### DISCUSSION TITLE
Adding some enums for cleaner code and improve consistency

### DISCUSSION THREAD

**Discussion by AsafMazuz1** [filamentphp/filament#6227] [Ideas]
Hey,
I want to add some enums for components Like BaseFileUpload.
I find it hard to remember the string in functions like $visibility.
I am thinking of adding an Enum folder with some Enum classes like:
Enum Visibility: string
{
    case PRIVATE = 'private';
    case PUBLIC = 'public';
}

And then, when I am using the component, I will be able to write it like that:
                Forms\Components\FileUpload::make('marker')
                    ->acceptedFileTypes(['image/*'])
                    ->maxFiles(1)
                    ->visibility(Visibility::PRIVATE)

This is just a simple example, 
and specific for Visabiltty, there is a League\Flysystem Visibility class that we can use,
I would like to know if it will be something worth doing for more components in the library.

I would be glad to know if this is something you would like to have on filamentphp, and if so, I will start working on it gladly :)

Thank you

**Comment by danharrin** [comment]
We cannot add these unfortunately, because our minimum PHP version is too low

**Reply by AsafMazuz1** [reply]
Ohh I see, 
I may add them as Classes.
Like that:
final class Visibility
{
    public const PUBLIC = 'public';
    public const PRIVATE = 'private';
}
And then we will still be able to use:
Visibility::PRIVATE.

What do you think?

**Reply by danharrin** [reply]
I don't think it's worth it for now until we support real enums. You can add such a class to your own project though.

**Reply by AsafMazuz1** [reply]
Ok, 
thank you,
I will find something else that I can do for the project :)
If you have something that an extra hand can help with,
please don't hesitate to let me know.
I would be glad to help :)
